content/articles/support-vector-regression-in-python/index.md

```diff
@@ -0,0 +1,303 @@
+
+```python
+y_p = y_p.reshape(-1,1)
+
+```
+Ouput:
+```bash
+[[  45000]
+ [  50000]
+ [  60000]
+ [  80000]
+ [ 110000]
+ [ 150000]
+ [ 200000]
+ [ 300000]
+ [ 500000]
+ [1000000]]
+
+```
+From the above output, y_p was successfully reshaped into a 2D array.
+
+The next thing we will do is to import the `StandardScalar` class and get our X_l and y_p variables scaled separately to the same range.
+
+```python
+from sklearn.preprocessing import StandardScaler
+StdS_X = StandardScaler()
+StdS_y = StandardScaler()
+X_l = StdS_X.fit_transform(X_l)
+y_p = StdS_y.fit_transform(y_p)
+
+```
+Let's simulteneously print and check if our two variiables were really scaled.
+```bash
+print("Scaled X_l:")
+print(X_l)
+print("Scaled y_p:")
+print(y_p)
+
+```
+Output:
+```bash
+Scaled X_l:
+[[-1.5666989 ]
+ [-1.21854359]
+ [-0.87038828]
+ [-0.52223297]
+ [-0.17407766]
+ [ 0.17407766]
+ [ 0.52223297]
+ [ 0.87038828]
+ [ 1.21854359]
+ [ 1.5666989 ]]
+Scaled y_p:
+[[-0.72004253]
+ [-0.70243757]
+ [-0.66722767]
+ [-0.59680786]
+ [-0.49117815]
+ [-0.35033854]
+ [-0.17428902]
+ [ 0.17781001]
+ [ 0.88200808]
+ [ 2.64250325]]
+
+```
+As we can see from the obtained output, both variables were scaled within the range -3 and +3.
+
+Our data is now ready to implement our SVR model. However, before we can do so, we will first visualize the data to know the nature of the SVR model that best fits it. So let us create a scatter plot of our two variables.
+
+```python
+plt.scatter(X_l, y_p, color = 'red') # plotting the training set
+plt.title('Scatter Plot') # adding a tittle to our plot
+plt.xlabel('Levels') # adds a label to the x-axis
+plt.ylabel('Salary') # adds a label to the y-axis
+plt.show() # prints
+
+```
+![Scatter plot](/engineering-education/support-vector-regression-in-python/plot0.png)
+
+The plot shows a nonlinear relationship between the `Levels` and the `Salary`. Therefore, we will use a kernelized version of the SVR model.
+
+### Implementing the Support Vector Regression
+
+To implement our model, first, we need to import it from the sk-learn and create an object to itself. Since we declared our data to be nonlinear, we will pass a kernel as `rbf`, which stands for, *Radial Basis function kernel*. After we have declared the kernel function, we will fit our data on the object. The following program performs these rules.
+
+```python
+# import the model
+from sklearn.svm import SVR
+ # create the model object
+regressor = SVR(kernel = 'rbf')
+# fit the model on the data
+regressor.fit(X_l, y_p)
+
+```
+Since the model is now ready, we can use it and make predictions. Let's see how we do so.
+```python
+A=regressor.predict(StdS_X.transform([[6.5]]))
+A
```

Comment:
Use print statements. The reader may or may not not be running in jupyter notebook cells